feat: Add Hyperpod Optimum-neuron LoRA example (#631)
aws-samples/awsome-distributed-training: 3.test_cases/pytorch/optimum-neuron/llama3/kubernetes/fine-tuning/README.md

> ## PEFT fine tuning of Llama 3 on SageMaker HyperPod with AWS Trainium

Reviewer: ```suggestion
## PEFT fine tuning of Llama 3 on Amazon EKS with AWS Trainium
```

Nothing HP specific

> This README demonstrates how to perform efficient supervised fine tuning for a Meta Llama 3.1 model using Parameter-Efficient Fine Tuning (PEFT) on AWS Trainium with SageMaker HyperPod. We use HuggingFace's Optimum-Neuron SDK to apply Low-Rank Adaptation (LoRA) to fine-tuning jobs, and use SageMaker HyperPod as the primary compute cluster to perform distributed training on Trainium.

Reviewer: ```suggestion
This README demonstrates how to perform efficient supervised fine tuning for a Meta Llama 3.1 model using Parameter-Efficient Fine Tuning (PEFT) on AWS Trainium with EKS. We use HuggingFace's Optimum-Neuron SDK to apply Low-Rank Adaptation (LoRA) to fine-tuning jobs, and use Trainiums to perform distributed trainin.
```

### Solution overview
This solution uses the following components:

> ### SageMaker HyperPod for distributed training infrastructure

Reviewer: ```suggestion
### Distributed training infrastructure
```

AWS Trainium chips for deep learning acceleration
Hugging Face Optimum-Neuron for integrating Trainium with existing models and tools
LoRA for parameter-efficient fine tuning

## 0. Prerequisites

### 0.1. EKS Cluster 
Before running this training, you'll need to create an Amazon EKS or a SageMaker HyperPod EKS cluster with at least 1 trn1.32xlarge/ trn1n.32xlarge. Instructions can be found in [1.architectures](../../1.architectures), the [aws-do-eks](https://bit.ly/do-eks) project, or the [eks-blueprints](https://github.com/aws-ia/terraform-aws-eks-blueprints) project.

### 0.2 Setup Persistant Volume Claim(PVC) for fsx 

We need to setup an PVC for FSx to store the tokenized data and training checkpoints. Please follow the link [here](https://catalog.workshops.aws/sagemaker-hyperpod-eks/en-US/01-cluster/06-fsx-for-lustre) to setup FSx CSI Driver and PVC. 

> 

Reviewer: HF_TOKEN access should be in the prerequisites.
